roles/workflowtemplate/tasks/main.yml

```diff
@@ -1,10 +1,37 @@
 ---
-- name: Read Secret Configuration
+- name: Read Secret Configuration from connection_secret
   kubernetes.core.k8s_info:
     kind: Secret
-    namespace: '{{ ansible_operator_meta.namespace }}'
+    api_version: v1
+    name: "{{ connection_secret }}"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+  register: tower_connection_secret
+  when: connection_secret is defined
+
+- name: DEPRECATED Read Secret Configuration from tower_auth_secret
+  kubernetes.core.k8s_info:
+    kind: Secret
+    api_version: v1
     name: "{{ tower_auth_secret }}"
-  register: tower_config_secret
+    namespace: "{{ ansible_operator_meta.namespace }}"
+  register: tower_connection_deprecated_secret
+  when: tower_auth_secret is defined
+
+- name: Set fact based on secret results
+  set_fact:
+     tower_config_secret: "{{ tower_connection_secret }}"
+  when: tower_connection_secret is not skipped and tower_connection_deprecated_secret is skipped
+
+- name: Set fact based on secret results
+  set_fact:
+    tower_config_secret: "{{ tower_connection_deprecated_secret }}"
+  when: tower_connection_deprecated_secret is not skipped and tower_connection_secret is skipped
```

Comment:
@cooktheryan These tasks only seem to affect the following task that checks if the secret-to-be-used exists.  

Don't we similarly need logic to set `connection_secret` to be the name of the secret to be used (`tower_auth_secret` or `connection_secret`) before job_definition.yml.j2 is applied?

Reply:
good catch...will patch up and add an extra test case as well 👍